### PR TITLE
Disable swipe when open modal

### PR DESCRIPTION
**Pull request by Jpablogs8**
- Fixes in the use of the X button when exiting the app
- Disabled swipe when opening other screens

### CODE REVIEW

**Comment by markgrossnickle** [comment]
@Jpablogs8 where is the fix for "Fixes in the use of the X button when exiting the app" ?

**Comment by Jpablogs8** [comment]
image

Mark, I accidentally uploaded that change to Development and then created a copy of Development, then reverted that commit to development and that's why it doesn't appear.

File: Scuti/Scripts/UI/Implementations/Modals/LogoutView.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            Close();
+            m_Callback?.Invoke(true);
+        }
+
+        void OnClickClose()
+        {
+            Close();
+            m_Callback?.Invoke(false);
+        }
```
Comment (markgrossnickle) [inline Scuti/Scripts/UI/Implementations/Modals/LogoutView.cs:46]: @Jpablogs8 doesn't this mean the callback will get called whether it is accepted or not? Should callback only be onClickClose?

File: Scuti/Scripts/UI/Navigation.cs [hunk 2]
```diff
@@ -63,11 +63,14 @@ public class BreadCrumbs
                 if (currentModal == v)
                 {
                     currentModal = null;
-
+                    UIManager.Offers.GetNavigator().isShowingCategories = true;
```
[the remark this answers is not in the document]
Comment (markgrossnickle) [reply Scuti/Scripts/UI/Navigation.cs:66]: @Jpablogs8 This should only be true if offers is showing. Currently if I am on the details page or cart and I close a modal I can then change categories again

File: Scuti/Scripts/UI/Utilities/ScrollRectEx.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.UI;
+using System;
+using UnityEngine.EventSystems;
+
+public class ScrollRectEx : ScrollRect
```
[the remark this answers is not in the document]
Comment (markgrossnickle) [reply Scuti/Scripts/UI/Utilities/ScrollRectEx.cs:7]: @Jpablogs8 is this class being used?

**Review by markgrossnickle — CHANGES_REQUESTED** [review]
Getting there!

File: Scuti/Scripts/UI/Navigation.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jpablogs8) [reply Scuti/Scripts/UI/Navigation.cs:66]: its true

File: Scuti/Scripts/UI/Utilities/ScrollRectEx.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jpablogs8) [reply Scuti/Scripts/UI/Utilities/ScrollRectEx.cs:7]: Deleted

File: Scuti/Scripts/UI/Implementations/Modals/LogoutView.cs (hunk 1 above)
Comment (Jpablogs8) [reply Scuti/Scripts/UI/Implementations/Modals/LogoutView.cs:46]: @markgrossnickle According to how I saw it working from Navigation you needed to know both cases to show the offers again. That's why the true or false

**Review by markgrossnickle — APPROVED** [review]
Looks good!

**Pull request merged**